### DISCUSSION TITLE
SPA

### DISCUSSION THREAD

**Discussion by danieludokike** [reactive-python/reactpy-django#165] [Question] [answered]
How do I achieve the SPA approach in react.js with django-reactpy. And I don't want my page to reload in submitting forms, or navigating through in the urls. This library is a powerful one but I can't find much on youtube and search

![page](https://github.com/reactive-python/reactpy-django/assets/68450806/e14ce46d-c709-4a4c-b730-befdf512f230)

**Comment by rmorshea** [comment, marked as answer]
Using the latest version of [`reactpy-router`](https://github.com/reactive-python/reactpy-router) in ReactPy Django is blocked on [this issue](https://github.com/reactive-python/reactpy/issues/1079). We plan on addressing this in the next week though.

**Reply by rmorshea** [reply]
With that said, `reactpy-router` [does not have a `form` component](https://github.com/reactive-python/reactpy-router/issues/7) yet. Probably not the easiest task for a new contributor to take on, but I'd be happy to help if you're interested.

**Reply by Archmonger** [reply]
Related issues:
- https://github.com/reactive-python/reactpy-django/issues/147
- https://github.com/reactive-python/reactpy-django/issues/36
